From 22e6eece284451c1d864b07fc788b88e3045f102 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Tue, 16 Sep 2014 17:04:34 -0400
Subject: [usb] core: Remove superfluous name casts

Message-id: <1410887463-20674-203-git-send-email-dzickus@redhat.com>
Patchwork-id: 95225
O-Subject: [RHEL7 PATCH 202/591] usb: core: Remove superfluous name casts
Bugzilla: 1110939
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1110939

commit 9f9af82ff3e3927d29474049675303284e1266f7
Author: Geert Uytterhoeven <geert@linux-m68k.org>
Date:   Tue Nov 12 20:07:22 2013 +0100

    usb: core: Remove superfluous name casts

    device_driver.name is "const char *"

    Signed-off-by: Geert Uytterhoeven <geert@linux-m68k.org>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/usb/core/driver.c b/drivers/usb/core/driver.c
index 47aade2..8d989b1 100644
--- a/drivers/usb/core/driver.c
+++ b/drivers/usb/core/driver.c
@@ -839,7 +839,7 @@ int usb_register_device_driver(struct usb_device_driver *new_udriver,
   return -ENODEV;
 
  new_udriver->drvwrap.for_devices = 1;
- new_udriver->drvwrap.driver.name = (char *) new_udriver->name;
+ new_udriver->drvwrap.driver.name = new_udriver->name;
  new_udriver->drvwrap.driver.bus = &usb_bus_type;
  new_udriver->drvwrap.driver.probe = usb_probe_device;
  new_udriver->drvwrap.driver.remove = usb_unbind_device;
@@ -900,7 +900,7 @@ int usb_register_driver(struct usb_driver *new_driver, struct module *owner,
   return -ENODEV;
 
  new_driver->drvwrap.for_devices = 0;
- new_driver->drvwrap.driver.name = (char *) new_driver->name;
+ new_driver->drvwrap.driver.name = new_driver->name;
  new_driver->drvwrap.driver.bus = &usb_bus_type;
  new_driver->drvwrap.driver.probe = usb_probe_interface;
  new_driver->drvwrap.driver.remove = usb_unbind_interface;
-- 
1.7.1